File: sms final assessment/stockmanagement/src/main/java/com/ustglobal/stockmanagement/service/OrderService.java
package com.ustglobal.stockmanagement.service;

import java.util.List;

import com.ustglobal.stockmanagement.dto.Order;



public interface OrderService {
	public boolean addOrder(Order order);
	public boolean updateOrder(Order order);
	public boolean deleteOrder(int order_id);
	public List<Order> getOrderbyId(int order_id);
	public List<Order> getAllOrder();

}

File: sms final assessment/front end/src/app/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  constructor(private http: HttpClient) { }


  getOrders(): Observable<any> {
    return this.http.get('http://localhost:8087/order/get-all');
  }
  deleteOrder(order_id: number): Observable<any> {
    return this.http.delete(`http://localhost:8087/order/delete/${order_id}`);
  }
  updateOrder(order_id: number, user): Observable<any> {
    return this.http.put(`http://localhost:8087/order/modify/${user.order_id}`, user);
  }
  addOrder(user): Observable<any> {
    return this.http.post('http://localhost:8087/order/add', user);
  }
}
